RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc

```diff
@@ -145,24 +146,26 @@ bool SiPixelTemplate2D::pushfile(int filenum, std::vector< SiPixelTemplateStore2
       
       if(theCurrentTemp.head.NTy != 0) {LOGERROR("SiPixelTemplate2D") << "Trying to load 1-d template info into the 2-d template object, check your DB/global tag!" << ENDL; return false;}
       
-      
-#ifdef SI_PIXEL_TEMPLATE_USE_BOOST
-      
       // next, layout the 2-d structure needed to store template
       
-      theCurrentTemp.entry.resize(boost::extents[theCurrentTemp.head.NTyx][theCurrentTemp.head.NTxx]);
-      
-#endif
-      
+      theCurrentTemp.entry = new SiPixelTemplateEntry2D*[theCurrentTemp.head.NTyx];

```

Comment:
I thought of smart pointers, but that can't happen for this PR, since it requires some thought, more testing, and stuff like that.  Can we do it next week (next PR) instead?

Reply:
The template store is created in pushfile(), which is a global static function.  They are destroyed in caller's destructor:
  http://cmslxr.fnal.gov/source/RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc#0110

So everything is fine.

Reply:
This is what I would like to test explicitely: the line you point out destroys 1D templates (```std::vector< SiPixelTemplateStore >```), while I don't see any destroyer of the 2D templates which are here at hand.